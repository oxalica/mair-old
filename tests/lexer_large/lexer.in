//! hugh?

use std::*;

/// just junk
#[test]
fn lexer_keyword_ident() {
    assert_eq!(lex("_"),        Ok(vec![(Ident("_"), 0..1)]));
    assert_eq!(lex("a"),        Ok(vec![(Ident("a"), 0..1)]));
    assert_eq!(lex("as"),       Ok(vec![(Keyword(Kw::As), 0..2)]));
    assert_eq!(lex("asc"),      Ok(vec![(Ident("asc"), 0..3)]));
    assert_eq!(lex("a0__c_"),   Ok(vec![(Ident("a0__c_"), 0..6)]));
    assert_eq!(lex("_9 a0"),    Ok(vec![(Ident("_9"), 0..2), (Ident("a0"), 3..5)]));
    assert_eq!(lex("macro_rules"), Ok(vec![(Ident("macro_rules"), 0..11)]));
}


impl<'input> Iterator for Lexer<'input> {
    /*!no real impl here*/
}

symbol_test!{ >>>=>>====>= }

