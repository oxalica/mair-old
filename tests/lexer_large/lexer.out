Ok([(InnerDoc(" hugh?\n"), "//! hugh?\n"), (Keyword(Use), "use"), (Ident("std"), "std"), (Symbol(ColonColon), "::"), (Symbol(Mul), "*"), (Symbol(Semi), ";"), (OuterDoc(" just junk\n"), "/// just junk\n"), (Symbol(Hash), "#"), (Delimiter { is_open: true, delim: Bracket }, "["), (Ident("test"), "test"), (Delimiter { is_open: false, delim: Bracket }, "]"), (Keyword(Fn), "fn"), (Ident("lexer_keyword_ident"), "lexer_keyword_ident"), (Delimiter { is_open: true, delim: Paren }, "("), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: true, delim: Brace }, "{"), (Ident("assert_eq"), "assert_eq"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("lex"), "lex"), (Delimiter { is_open: true, delim: Paren }, "("), (Literal(StrLike { is_bytestr: false, s: "_" }), "\"_\""), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Ident("Ok"), "Ok"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("vec"), "vec"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Bracket }, "["), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("Ident"), "Ident"), (Delimiter { is_open: true, delim: Paren }, "("), (Literal(StrLike { is_bytestr: false, s: "_" }), "\"_\""), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Literal(IntLike { ty: None, val: 0 }), "0"), (Symbol(DotDot), ".."), (Literal(IntLike { ty: None, val: 1 }), "1"), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: false, delim: Bracket }, "]"), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Semi), ";"), (Ident("assert_eq"), "assert_eq"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("lex"), "lex"), (Delimiter { is_open: true, delim: Paren }, "("), (Literal(StrLike { is_bytestr: false, s: "a" }), "\"a\""), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Ident("Ok"), "Ok"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("vec"), "vec"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Bracket }, "["), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("Ident"), "Ident"), (Delimiter { is_open: true, delim: Paren }, "("), (Literal(StrLike { is_bytestr: false, s: "a" }), "\"a\""), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Literal(IntLike { ty: None, val: 0 }), "0"), (Symbol(DotDot), ".."), (Literal(IntLike { ty: None, val: 1 }), "1"), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: false, delim: Bracket }, "]"), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Semi), ";"), (Ident("assert_eq"), "assert_eq"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("lex"), "lex"), (Delimiter { is_open: true, delim: Paren }, "("), (Literal(StrLike { is_bytestr: false, s: "as" }), "\"as\""), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Ident("Ok"), "Ok"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("vec"), "vec"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Bracket }, "["), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("Keyword"), "Keyword"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("Kw"), "Kw"), (Symbol(ColonColon), "::"), (Ident("As"), "As"), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Literal(IntLike { ty: None, val: 0 }), "0"), (Symbol(DotDot), ".."), (Literal(IntLike { ty: None, val: 2 }), "2"), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: false, delim: Bracket }, "]"), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Semi), ";"), (Ident("assert_eq"), "assert_eq"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("lex"), "lex"), (Delimiter { is_open: true, delim: Paren }, "("), (Literal(StrLike { is_bytestr: false, s: "asc" }), "\"asc\""), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Ident("Ok"), "Ok"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("vec"), "vec"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Bracket }, "["), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("Ident"), "Ident"), (Delimiter { is_open: true, delim: Paren }, "("), (Literal(StrLike { is_bytestr: false, s: "asc" }), "\"asc\""), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Literal(IntLike { ty: None, val: 0 }), "0"), (Symbol(DotDot), ".."), (Literal(IntLike { ty: None, val: 3 }), "3"), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: false, delim: Bracket }, "]"), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Semi), ";"), (Ident("assert_eq"), "assert_eq"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("lex"), "lex"), (Delimiter { is_open: true, delim: Paren }, "("), (Literal(StrLike { is_bytestr: false, s: "a0__c_" }), "\"a0__c_\""), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Ident("Ok"), "Ok"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("vec"), "vec"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Bracket }, "["), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("Ident"), "Ident"), (Delimiter { is_open: true, delim: Paren }, "("), (Literal(StrLike { is_bytestr: false, s: "a0__c_" }), "\"a0__c_\""), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Literal(IntLike { ty: None, val: 0 }), "0"), (Symbol(DotDot), ".."), (Literal(IntLike { ty: None, val: 6 }), "6"), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: false, delim: Bracket }, "]"), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Semi), ";"), (Ident("assert_eq"), "assert_eq"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("lex"), "lex"), (Delimiter { is_open: true, delim: Paren }, "("), (Literal(StrLike { is_bytestr: false, s: "_9 a0" }), "\"_9 a0\""), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Ident("Ok"), "Ok"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("vec"), "vec"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Bracket }, "["), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("Ident"), "Ident"), (Delimiter { is_open: true, delim: Paren }, "("), (Literal(StrLike { is_bytestr: false, s: "_9" }), "\"_9\""), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Literal(IntLike { ty: None, val: 0 }), "0"), (Symbol(DotDot), ".."), (Literal(IntLike { ty: None, val: 2 }), "2"), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("Ident"), "Ident"), (Delimiter { is_open: true, delim: Paren }, "("), (Literal(StrLike { is_bytestr: false, s: "a0" }), "\"a0\""), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Literal(IntLike { ty: None, val: 3 }), "3"), (Symbol(DotDot), ".."), (Literal(IntLike { ty: None, val: 5 }), "5"), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: false, delim: Bracket }, "]"), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Semi), ";"), (Ident("assert_eq"), "assert_eq"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("lex"), "lex"), (Delimiter { is_open: true, delim: Paren }, "("), (Literal(StrLike { is_bytestr: false, s: "macro_rules" }), "\"macro_rules\""), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Ident("Ok"), "Ok"), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("vec"), "vec"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Bracket }, "["), (Delimiter { is_open: true, delim: Paren }, "("), (Ident("Ident"), "Ident"), (Delimiter { is_open: true, delim: Paren }, "("), (Literal(StrLike { is_bytestr: false, s: "macro_rules" }), "\"macro_rules\""), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Comma), ","), (Literal(IntLike { ty: None, val: 0 }), "0"), (Symbol(DotDot), ".."), (Literal(IntLike { ty: None, val: 11 }), "11"), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: false, delim: Bracket }, "]"), (Delimiter { is_open: false, delim: Paren }, ")"), (Delimiter { is_open: false, delim: Paren }, ")"), (Symbol(Semi), ";"), (Delimiter { is_open: false, delim: Brace }, "}"), (Keyword(Impl), "impl"), (Symbol(Lt), "<"), (Lifetime("input"), "\'input"), (Symbol(Gt), ">"), (Ident("Iterator"), "Iterator"), (Keyword(For), "for"), (Ident("Lexer"), "Lexer"), (Symbol(Lt), "<"), (Lifetime("input"), "\'input"), (Symbol(Gt), ">"), (Delimiter { is_open: true, delim: Brace }, "{"), (InnerDoc("no real impl here"), "/*!no real impl here*/"), (Delimiter { is_open: false, delim: Brace }, "}"), (Ident("symbol_test"), "symbol_test"), (Symbol(Bang), "!"), (Delimiter { is_open: true, delim: Brace }, "{"), (Symbol(Shr), ">>"), (Symbol(Ge), ">="), (Symbol(ShrEq), ">>="), (Symbol(EqEq), "=="), (Symbol(RFatArrow), "=>"), (Symbol(Eq), "="), (Delimiter { is_open: false, delim: Brace }, "}")])
